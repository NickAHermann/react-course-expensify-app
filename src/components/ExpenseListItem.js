// Export a stateless functional component
// description, amount, createdAt

import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3>
      <Link to={`/edit/${id}`}> {description}</Link>
    </h3>
    <h5>
      {amount} - {createdAt}
    </h5>
  </div>
);

export default ExpenseListItem;
