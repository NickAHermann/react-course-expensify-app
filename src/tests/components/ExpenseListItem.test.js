import React from "react";

import moment from "moment";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";

const expenses = [
  {
    id: "1",
    description: "Gum",
    note: "",
    amount: 195,
    createdAt: 0
  },
  {
    id: "2",
    description: "Rent",
    note: "",
    amount: 10000,
    createdAt: moment(0)
      .subtract(4, "days")
      .valueOf()
  },
  {
    id: "3",
    description: "credit card",
    note: "",
    amount: 4444,
    createdAt: moment(0)
      .add(4, "days")
      .valueOf()
  }
];

test("should render an instance of ExpenseListItem", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
  expect(wrapper).toMatchSnapshot();
});
