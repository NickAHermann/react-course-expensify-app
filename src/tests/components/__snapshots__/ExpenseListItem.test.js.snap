// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an instance of ExpenseListItem 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/2"
    >
       
      Rent
    </Link>
  </h3>
  <h5>
    10000
     - 
    -345600000
  </h5>
</div>
`;
